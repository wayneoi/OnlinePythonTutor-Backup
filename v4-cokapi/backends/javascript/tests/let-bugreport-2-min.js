// minimized form of let-bugreport-2.js
function multiply(arr, n) {
  if (n<=0) {
    return arr[0];
  } else {
    let result = multiply(arr, n-1) * arr[n];
    return result;
  }
}

multiply([2,3], 1);
