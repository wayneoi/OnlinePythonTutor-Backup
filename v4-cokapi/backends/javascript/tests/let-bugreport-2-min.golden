{ code: '// minimized form of let-bugreport-2.js\nfunction multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}\n\nmultiply([2,3], 1);',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 11,
       col: 0,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 1 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 3,
       col: 2,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', 'result (block 1)' ],
            encoded_locals: 
             { arr: [ 'REF', 2 ],
               n: 1,
               'result (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 6,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: false,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', 'result (block 1)' ],
            encoded_locals: 
             { arr: [ 'REF', 2 ],
               n: 1,
               'result (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } },
          { func_name: 'multiply',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f3',
            ordered_varnames: [ 'arr', 'n' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 0 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 3,
       col: 2,
       event: 'call',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: false,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', 'result (block 1)' ],
            encoded_locals: 
             { arr: [ 'REF', 2 ],
               n: 1,
               'result (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } },
          { func_name: 'multiply',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f3',
            ordered_varnames: [ 'arr', 'n' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 0 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 4,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: false,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', 'result (block 1)' ],
            encoded_locals: 
             { arr: [ 'REF', 2 ],
               n: 1,
               'result (block 1)': [ 'JS_SPECIAL_VAL', 'undefined' ] } },
          { func_name: 'multiply',
            frame_id: 3,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f3',
            ordered_varnames: [ 'arr', 'n', '__return__' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 0, __return__: 2 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 4,
       col: 0,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', 'result (block 1)' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 1, 'result (block 1)': 6 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 7,
       col: 4,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: 'multiply',
       stack_to_render: 
        [ { func_name: 'multiply',
            frame_id: 2,
            is_highlighted: true,
            is_parent: false,
            is_zombie: false,
            parent_frame_id_list: [],
            unique_hash: 'multiply_f2',
            ordered_varnames: [ 'arr', 'n', '__return__' ],
            encoded_locals: { arr: [ 'REF', 2 ], n: 1, __return__: 6 } } ],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 7,
       col: 0,
       event: 'return',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ],
          '2': [ 'LIST', 2, 3 ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { multiply: [ 'REF', 1 ] },
       ordered_globals: [ 'multiply' ],
       line: 11,
       col: 19,
       event: 'step_line',
       heap: 
        { '1': 
           [ 'JS_FUNCTION',
             'multiply',
             'function multiply(arr, n) {\n  if (n<=0) {\n    return arr[0];\n  } else {\n    let result = multiply(arr, n-1) * arr[n];\n    return result;\n  }\n}',
             null,
             null ] } } ] }
